src/connectors/onchain_events/mod.rs

```diff
@@ -151,7 +191,7 @@ impl RealL1Client {
 }
 
 #[async_trait]
-impl L1Client for RealL1Client {
+impl ChainAPI for RealL1Client {
     async fn resolve_ens_name(&self, name: String) -> Result<Address, EnsError> {
         foundry_common::ens::NameOrAddress::Name(name)
```

Comment:
Does this work? I thought that this hard codes the address for the ens resolver on eth. Does the basename resolver use the same address?

Reply:
Yeah, you're right. Fixed by bringing in the foundry validation code here.